packages/shared/whenever/index.ts

```diff
@@ -12,16 +13,25 @@ export interface WheneverOptions extends WatchOptions {
   once?: boolean
 }
 
+// overloads
+export function whenever<T extends Readonly<WatchSource<unknown>[]>>(sources: readonly [...T] | T, cb: WatchCallback<MapSources<T>, MapOldSources<T, true>>, options?: WheneverOptions): WatchHandle
+export function whenever<T>(source: WatchSource<T | false | null | undefined>, cb: WatchCallback<T>, options?: WheneverOptions): WatchHandle
+export function whenever<T extends object>(source: T, cb: WatchCallback<T>, options?: WheneverOptions): WatchHandle
```

Comment:
what is this one accounting for? do you have an example, or can you point me at the test that covers this one?